lib/mcrain/riak.rb

```diff
@@ -1,27 +1,84 @@
+# -*- coding: utf-8 -*-
 require 'mcrain'
 
 # require 'riak'
+require 'net/http'
 
 module Mcrain
   class Riak < Base
 
-    class << self
-      # path to clone of https://github.com/hectcastro/docker-riak
-      attr_accessor :docker_riak_path
+    self.server_name = :riak
+
+    self.container_image = "hectcastro/riak"
+
+    attr_accessor :automatic_clustering
+    attr_writer :cluster_size, :backend
+    def cluster_size
+      @cluster_size ||= 1
+    end
+    def backend
+      @backend ||= "bitcask" # "leveldb"
     end
 
-    self.server_name = :riak
+    # docker run -e "DOCKER_RIAK_CLUSTER_SIZE=${DOCKER_RIAK_CLUSTER_SIZE}" \
+    #            -e "DOCKER_RIAK_AUTOMATIC_CLUSTERING=${DOCKER_RIAK_AUTOMATIC_CLUSTERING}" \
+    #            -e "DOCKER_RIAK_BACKEND=${DOCKER_RIAK_BACKEND}" \
+    #            -p $publish_http_port \
+    #            -p $publish_pb_port \
+    #            --link "riak01:seed" \
+    #            --name "riak${index}" \
+    #            -d hectcastro/riak > /dev/null 2>&1
+    class Node
+      include ContainerController
+
+      self.container_image = "hectcastro/riak"
+
+      self.port = 8087 # protocol buffer
+      # self.http_port = 8098 # HTTP
+
+      attr_reader :owner
+      attr_accessor :primary_node
+      def initialize(owner)
+        @owner = owner
+      end
+
+      def http_port
+        @http_port ||= find_portno
+      end
 
-    self.container_image = nil # not use docker directly
-    self.port = 8087
+      def build_docker_options
+        r = super
+        r['HostConfig']['PortBindings']["8098/tcp"] = [{ 'HostPort': http_port.to_s }]
```

Comment:
`[{ 'HostPort': http_port.to_s }]` should be `[{ 'HostPort' => http_port.to_s }]`